sdk/src/Core/AWSXRayRecorder.netcore.cs

```diff
@@ -83,19 +86,22 @@ public static void InitializeInstance(IConfiguration configuration = null, AWSXR
             XRayOptions xRayOptions = XRayConfiguration.GetXRayOptions(configuration);
             var recorderBuilder = GetBuilder(xRayOptions);
 
-            if (recorder != null)
-            {
-                _logger.DebugFormat("Using custom X-Ray recorder.");
-                recorder.XRayOptions = xRayOptions;
-                recorder = recorderBuilder.Build(recorder);
-            }
-            else
-            {
-                _logger.DebugFormat("Using default X-Ray recorder.");
-                recorder = recorderBuilder.Build(xRayOptions);
+            if (recorder != null)
+            {
+                _logger.DebugFormat("Using custom X-Ray recorder.");
+                recorder.XRayOptions = xRayOptions;
+                recorder = recorderBuilder.Build(recorder);
+            }
+            else
+            {
+                _logger.DebugFormat("Using default X-Ray recorder.");
+                recorder = recorderBuilder.Build(xRayOptions);
+            }
+
+            lock (_instance_lock)
```

Comment:
I don't think we need a lock here `InitializeInstance()` is called only once during configuring the web app.